src/adv.py

```diff
@@ -49,3 +61,118 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+
+def print_items(item_array):
+
+    if len(item_array) > 0:
+        print('Items in the Room:')
+        for item in item_array:
+            print(item)
+    else:
+        print('No items in Room.')
+
+
+#enter your name here
+player_name = input('Before you proceed, please type in a name for your player! ')
+
+
+# Initial room variable:
+current_room = room['outside']
+
+
+# initializes the Player class with user inputted name and initial room (cave entrance)
+character = Player(player_name, room['outside'])
+
+while True:
+
+    print(
+        f'\nCurrent room: name {current_room.name}, description {current_room.description}\n')
+
+    print_items(current_room.items)
+
+    user_input = input(
+        'Please enter a direction you want the player to move. Selections include: [n, s, e, w]! ')
+
+    user_input = user_input.split(' ')
+
+    if len(user_input) == 1:
+
+        if user_input[0] == 'n':
+
+            if not isinstance(current_room.n_to, str):
```

Comment:
Good work validating the commands. How could we better compose this repeated functionality?